src/transformers/testing_utils.py

```diff
@@ -1570,6 +1572,34 @@ def LoggingLevel(level):
         transformers_logging.set_verbosity(orig_level)
 
 
+class TemporaryHubRepo:
+    """
+    This is a context manager to temporarily change transformers modules logging level to the desired value and have it
+    restored to the original setting at the end of the scope.
+
+    Example:
+
+    ```python
+    with LoggingLevel(logging.INFO):
+        AutoModel.from_pretrained("openai-community/gpt2")  # calls logger.info() several times
+    ```
+    """
+    def __init__(self, prefix="repo", random_id=None, token=None):
```

Comment:
I should prepare a `random_id` if it is not specified in the caller.

I would prefer to use `tempfile.TemporaryDirectory` way to create a random id (directory name) though, but it's not really necessary.

My final goal is actually to have something


```python
def __init__(self, prefix="repo", random_id=None, token=None, create_local_dir=False):
```
which will use `tempfile.TemporaryDirectory` to create local temp dir if specified as `True`, so we don't have to do

```python
        with tempfile.TemporaryDirectory() as tmp_dir:
            with TemporaryHubRepo(prefix=repo_id, random_id=Path(tmp_dir).name, token=self._token) as tmp_repo:
```

but probably it's not a good idea?

Reply:
in `huggingface_hub`, we use 

```py
def repo_name(id: Optional[str] = None, prefix: str = "repo") -> str:
    """
    Return a readable pseudo-unique repository name for tests.

    Example:
    ```py
    >>> repo_name()
    repo-2fe93f-16599646671840

    >>> repo_name("my-space", prefix='space')
    space-my-space-16599481979701
    """
    if id is None:
        id = uuid.uuid4().hex[:6]
    ts = int(time.time() * 10e3)
    return f"{prefix}-{id}-{ts}"
```

to make the repo name pseudo-unique. To be honest, `prefix: str = "repo"` is there for legacy reasons but it doesn't bring anything useful to the user. What I would advice you to do is to remove both `random_id` and `prefix` from this helper to simplify things as much as possible for the users.